server/src/main/java/com/google/coffeehouse/servlets/CreatePersonServlet.java

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.servlets;
+
+import com.google.coffeehouse.common.Person;
+import com.google.coffeehouse.util.IdentifierGenerator;
+import com.google.coffeehouse.util.UuidWrapper;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet to create a {@link Person} from URL parameter input, 
+ * save it in database, and return it in JSON format.
+ */
+@WebServlet("/create-person")
+public class CreatePersonServlet extends HttpServlet {
+
+  /** 
+   * The error string sent by the response object in doPost when no email or nickname parameter 
+   * exists.
+   */
+  public static final String EMAIL_OR_NICKNAME_ERROR = "- email or nickname not specified.";
+  private IdentifierGenerator idGen = null;
+  
+  /** 
+   * Overloaded constructor for dependency injection.
+   * @param idGen the {@link IdentifierGenerator} that is used when constructing the Person
+   */
+  public CreatePersonServlet(IdentifierGenerator idGen) {
+    super();
+    this.idGen = idGen;
+  }
+
+  /** 
+   * Explicit default constructor used for instantiating the servlet when not testing.
+   */
+  public CreatePersonServlet() {
+    super();
+  }
+
+  /** 
+   * Creates a {@link Person} object, saves it in the database and returns it in JSON format.
+   * @param request the post request that must have {@code email} and {@code nickname} defined
+   *     as parameters or the respone will send a "bad request error"
```

Comment:
Nit: respone -> response.
Consider adding a note of what the standard status code is for this error.

Reply:
Both have been fixed, thanks for catching that!